gammapy/irf/effective_area.py

```diff
@@ -246,55 +246,32 @@ def max_area(self):
         cleaned_data = self.data.data[np.where(~np.isnan(self.data.data))]
         return cleaned_data.max()
 
-    def find_energy(self, aeff, reverse=False):
+    def find_energy(self, aeff, emin=None, emax=None):
```

Comment:
We the new implementation, we rely on `brentq` to find a root but we don't control where it will be located i.e. closest to lower bound or higher bound. Is this an issue? As far as I can see the `reverse = True` option is used in `SpectrumObservation.compute_energy_threshold` .

Reply:
This is indeed an issue, that I wasn't sure how to deal with. The general problem is that `bias` as well as `aeff` are not necessarily invertable, because of noise or irregular shapes. I think it's really hard to find an algorithm that does it right for all cases. The old solution was suffering from this problem as well. Here are two examples from the test datasets we use:

Effective area:
![aeff](https://user-images.githubusercontent.com/3715928/53089658-77aa3080-350d-11e9-9e4b-9d589ea7ea3d.png)

In this case, let's say if you look for the 90% effective area from the high energy end, there are basically three solutions and you don't know which one to choose...


Energy dispersion:

![edisp](https://user-images.githubusercontent.com/3715928/53090527-fdc77680-350f-11e9-9820-8b409516d1b5.png)
 
In this case, when you look for the energy where the bias is `~3.1` it's again not unique. At the high energy end its even worse, because of noise.

In both cases I thought the best you could do is to provide the `emin` and `emax` arguments, that users can define the search interval (e.g. by looking at those plots first...). I guess in practice one can find reasonable default values for `emin` and `emax` (e.g. peak bias or peak aeff) such that finding the lower energy threshold is stable in 99% of the cases. For the high energy end I don't really know, if it's reasonable at all to work with the bias and aeff criteria. I'm happy for any suggestion how to deal with this... 

CC @lmohrmann

Reply:
This is indeed not a simple problem, because you never know what the curves may look like. I like the idea of using a restricted energy range -- ideally we would provide default values for `emin` and `emax` that work well in most cases (maybe you tried that already, see my other comment).

The question whether the high-energy bound should be determined from the effective area / bias histogram at all or if something else would be better is also valid. I don't have a good suggestions for an alternative definition, though. Simply use the energy up to which the effective area is provided?!